sdks/java/core/src/main/java/org/apache/beam/sdk/util/HistogramData.java

```diff
@@ -184,6 +193,26 @@ public synchronized void record(double value) {
       buckets[bucketType.getBucketIndex(value)]++;
       numBoundedBucketRecords++;
     }
+    updateStatistics(value);
+  }
+
+  /**
+   * Update 'mean' and 'sum of squared deviations' statistics with the newly recorded value <a
+   * href="https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm">
+   * Welford Method</a>.
+   *
+   * @param value
+   */
+  private void updateStatistics(double value) {
```

Comment:
nit: mark as synchronized to guard against a non-threadsafe/non synchronized caller

Reply:
done